src/lib/features/project-environments/environment-store-type.ts

```diff
@@ -30,4 +30,7 @@ export interface IEnvironmentStore extends Store<IEnvironment, string> {
         projectId: string,
         query?: Object,
     ): Promise<IProjectEnvironment[]>;
+    getChangeRequestEnvironments(
```

Comment:
tells us which of the environments that we asked for have change requests enabled and how many approvals are needed

Reply:
so this only returns envs that **have** CRs enabled. Non-CR envs are filtered out, right? Is there ever a use case where we'd want both? Maybe not, but I was a little surprised that it would only return envs with CRs. Maybe the name's a bit weird because it's "**get**", but you pass in a list that it filters? 🤷🏼 This is not important, by the way. Feel free to disregard it 😄 

And finally: is it more useful to return a list of items or an object that has `{ [name]: requiredApprovals }` instead? Again, not important. If you need it in that format, you can always reduce it later, I guess 💁🏼

Reply:
TBH This method was use-case driven. So I literally designed the method API from a service. I was thinking of the ideal method I'd like to have. And it lead to the method where you tell the list of envs and get only those that have change requests. We could also call it filter

Reply:
Yeh, all good. It made sense the further into the PR I got. There's no need to make a "theoretically perfect" method if it doesn't serve your use case. Totes fine to leave it.